src/EnergyPlus/api/plugin.py

```diff
@@ -62,6 +62,9 @@ def on_begin_zone_timestep_after_init_heat_balance(self, state) -> int:
     def on_begin_timestep_before_predictor(self, state) -> int:
         pass
 
+    def on_begin_zone_timestep_before_set_current_weather(self, state) -> int:
```

Comment:
Add the new calling point to the Python EMS Plugin base class, allowing users to override the function and EnergyPlus will then call the overridden version in their derived class instance.